Original patch from: ../crosstool-NG/ct-ng.trunk/patches/binutils/2.18.50.0.4/120-check-ldrunpath-length.patch

-= BEGIN original header =-
Check LD_RUN_PATH length, copied from buildroot.

 binutils-2.18.50.0.6/ld/emultempl/elf32.em |    4     4     0     0 ++++
 1 file changed, 4 insertions(+)

-= END original header =-

diff -durN binutils-2.18.50.0.4.orig/ld/emultempl/elf32.em binutils-2.18.50.0.4/ld/emultempl/elf32.em
--- binutils-2.18.50.0.4.orig/ld/emultempl/elf32.em	2008-02-08 17:44:55.000000000 +0100
+++ binutils-2.18.50.0.4/ld/emultempl/elf32.em	2009-03-07 12:17:31.000000000 +0100
@@ -1220,6 +1220,8 @@
 	      && command_line.rpath == NULL)
 	    {
 	      lib_path = (const char *) getenv ("LD_RUN_PATH");
+	      if ((lib_path) && (strlen (lib_path) == 0))
+		  lib_path = NULL;
 	      if (gld${EMULATION_NAME}_search_needed (lib_path, &n,
 						      force))
 		break;
@@ -1404,6 +1406,8 @@
   rpath = command_line.rpath;
   if (rpath == NULL)
     rpath = (const char *) getenv ("LD_RUN_PATH");
+  if ((rpath) && (strlen (rpath) == 0))
+      rpath = NULL;
   if (! (bfd_elf_size_dynamic_sections
 	 (output_bfd, command_line.soname, rpath,
 	  command_line.filter_shlib,
